File: binds/go.v1/src/repoclient.go
package main

import (
	"fmt"
	"time"

	"gopkg.in/inconshreveable/log15.v2"
	"gopkg.in/project-iris/iris-go.v1"
)

func init() {
	iris.Log.SetHandler(log15.DiscardHandler())
}

func main() {
	// Connect to the Iris network
	connection, err := iris.Connect(55555)
	if err != nil {
		panic(err)
	}
	defer connection.Close()

// START OMIT
// Open an outbound tunnel to a data store
tunnel, err := connection.Tunnel("repository", time.Second) // HLtunnel
if err != nil {
	fmt.Println("Tunneling failed:", err); return
}
defer tunnel.Close() // HLtunnel

// Request a file and retrieve the multi-part response
tunnel.Send([]byte("some file"), time.Second) // HLtunnel
for {
	msg, err := tunnel.Recv(time.Second) // HLtunnel
	if err != nil {
		break
	}
	fmt.Println(string(msg))
}
// END OMIT
}

File: binds/erlang.v1/src/repoclient.erl
-module(repoclient).
-export([main/0]).

%% START OMIT
main() ->
    {ok, Client} = iris_client:start(55555),

    % Open an outbound tunnel to a data store
    case iris_client:tunnel(Client, "repository", 1000) of // HLtunnel
        {error, Reason} -> io:format("Tunneling failed: ~p~n", [Reason]);
        {ok, Tunnel}    ->
            % Request a file and retrieve the multi-part response
            iris_tunnel:send(Tunnel, <<"some file">>, 1000), // HLtunnel
            fetch(Tunnel)
    end,
    ok = iris_client:stop(Client).

fetch(Tunnel) ->
	case iris_tunnel:recv(Tunnel, 1000) of // HLtunnel
		{ok, Data} -> io:format("~s~n", [Data]), fetch(Tunnel);
		_Otherwise -> ok = iris_tunnel:close(Tunnel) // HLtunnel
	end.
%% END OMIT
